src/Cthulhu.jl

```diff
@@ -434,10 +435,17 @@ function _descend(term::AbstractTerminal, interp::AbstractInterpreter, curs::Abs
         """)
     end
     while true
-        if override !== nothing
+        if override !== nothing && !isa(override, SemiConcreteCallInfo)
             (; src, rt, infos, slottypes, codeinf, effects) = lookup_constproped(interp, curs, override, optimize)
         else
-            if optimize
+            if isa(override, SemiConcreteCallInfo)
```

Comment:
We should lift this branch up, since otherwise the `lookup` call later will override the IR.